source/simulator/entropy_viscosity.cc

```diff
@@ -428,13 +429,105 @@ namespace aspect
                                                    scratch.finite_element_values.get_mapping(),
                                                    scratch.material_model_outputs);
 
+        if (parameters.use_supg == false)
+          {
         viscosity_per_cell[cell->active_cell_index()] = compute_viscosity(scratch,
                                                                           global_max_velocity,
                                                                           global_field_range.second - global_field_range.first,
                                                                           0.5 * (global_field_range.second + global_field_range.first),
                                                                           global_entropy_variation,
                                                                           cell->diameter(),
                                                                           advection_field);
+          }
+        else
+          {
+            // RG: remove me
+        //nu = 1e-3;
+        // Compute norm of advection field
+        double norm_of_advection_field = 0.0;
+        // things needed for the calculating of tau for SUPG loop
+        double max_conductivity_on_cell = 0.0;
+
+        for (unsigned int q=0; q<n_q_points; ++q)
+          {
+            if (advection_field.is_temperature())
+              {
+                norm_of_advection_field =
+                std::max(scratch.current_velocity_values[q].norm()*
+                         (scratch.material_model_outputs.densities[q] *
+                          scratch.material_model_outputs.specific_heat[q] +
+                          scratch.heating_model_outputs.lhs_latent_heat_terms[q]),
+                         norm_of_advection_field);
+
+                max_conductivity_on_cell =
+                std::max(scratch.material_model_outputs.thermal_conductivities[q],max_conductivity_on_cell);
+              }
+            else
+              {
+                norm_of_advection_field =
+                std::max(scratch.current_velocity_values[q].norm(),norm_of_advection_field);
+
+                max_conductivity_on_cell = 0.0;
+              }
+          }
+
+        double fe_order
+          = (advection_field.is_temperature()
+             ?
+             parameters.temperature_degree
+             :
+             parameters.composition_degree
+            );
+
+        // Compute tau for SUPG
+        if (parameters.use_supg == true)
+          {
+            double h = cell->diameter();
+            //nu = .1*h*h*norm_of_advection_field; //RG: temp
+            if (norm_of_advection_field>1e-8)
```

Comment:
this messy logic needs to be rewritten. I am also convinced these hard-coded 1e-8 won't work for anything dimensional.